### DISCUSSION TITLE
What is the best way to keep Oqtane latest version

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1612] [Q&A] [answered]
Hi,

I've created the clone of Oqtane's Client, Server & Shared projects. All these projects mostly contains UI part (ActionLink, ActionDialog, etc.,) of the Oqtane framework. If I required any UI changes with functionalities of the Oqtane then I can copy and paste razor component into my project and do the changes. Also, I can publish this projects like as module as a dll to the Oqtane's Server project (bin folder). Is this the best way to keep Oqtane latest version?

**Comment by sbwalker** [comment, marked as answer]
Oqtane is a framework - not an application template or a boilerplate application. The general rule with any framework is to never make your own custom changes to the framework itself, as if you do, you will be responsible for the effort of maintaining your own fork forever. Frameworks provide extensibility options so that you can integrate your own custom functionality without modifying the framework. This is the purpose of modules and themes in Oqtane.

**Comment by leigh-pointer** [comment]
> Hi,
> 
> I've created the clone of Oqtane's Client, Server & Shared projects. All these projects mostly contains UI part (ActionLink, ActionDialog, etc.,) of the Oqtane framework. If I required any UI changes with functionalities of the Oqtane then I can copy and paste razor component into my project and do the changes. Also, I can publish this projects like as module as a dll to the Oqtane's Server project (bin folder). Is this the best way to keep Oqtane latest version?

@Sharafudeen have a quick watch at [https://www.youtube.com/watch?v=JPfUZPlRRCE](https://www.youtube.com/watch?v=JPfUZPlRRCE) it is a very good place to start.

**Reply by Sharafudeen** [reply]
@leigh-pointer Thanks for sharing. I've already started this and it is very good. I just wanted to know whether my approach is good or any best practices out there.

**Comment by horacioj** [comment]
Related question:

Let's say I'm implementing  `ServerStartup : IServerStartup`  in a module to register a required service in `ConfigureServices`. 

It is OK to add my custom settings to the Oqtane's \Oqtane.Server\appsettings.json (to read them using something like `var appSettings = sp.GetService<IConfiguration>().GetSection("mymodulestartupsettings");` ? Or is there a way to keep the provided appsettings.json unmodified?

**Reply by sbwalker** [reply]
Yes you can add settings to appsettings.json - configuration files are expected to be modified.
